pkg/util/azureclient/typeToCode.go

```diff
@@ -0,0 +1,33 @@
+package azureclient
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import "strings"
+
+// GetCodeFromType gets the code (that keys APIVersions) from a resource type
+func GetCodeFromType(_type string) string {
```

Comment:
I agree that we need to be able to map from a received _type to a version, but I'd rather not do it like this.  Can you refactor so that APIVersions looks like something this, then work from there?

```
var APIVersions = map[string]string{
	"Microsoft.Authorization":                 "2018-09-01-preview",
	"Microsoft.Authorization/denyAssignments": "2018-07-01-preview",
	"Microsoft.Authorization/roleDefinitions": "2018-01-01-preview",
	"Microsoft.Compute":                       "2019-03-01",
	"Microsoft.ContainerRegistry":             "2019-05-01",
	"Microsoft.DocumentDB":                    "2019-08-01",
	"Microsoft.Insights":                      "2018-03-01",
	"Microsoft.KeyVault":                      "2016-10-01",
	"Microsoft.ManagedIdentity":               "2018-11-30",
	"Microsoft.Network":                       "2019-07-01",
	"Microsoft.Network/dnsZones":              "2018-05-01",
	"Microsoft.Network/privateDnsZones":       "2018-09-01",
	"Microsoft.Storage":                       "2019-04-01",
}
```